### PR TITLE
feat: add test page that hits an endpoint that logs request headers

### PR DESCRIPTION
**Pull request by johnnyreilly**
Related to https://github.com/arethetypeswrong/arethetypeswrong.github.io/pull/76#issuecomment-1694579703 - this PR adds a temporary page to ATTW called test.html. It has an open graph image that points to this: https://github.com/johnnyreilly/blog.johnnyreilly.com/blob/main/blog-website/api/image-with-text/index.ts

Which serves up this image: https://johnnyreilly.com/api/image-with-text
and logs the request headers to an application insights instance like so:

image

Are you up for merging this so we can see what comes through when we hit https://arethetypeswrong.github.io/test.html ?

Unfortunately, GitHub Pages don't support fallback files, which means we need a querystring to show up in order to make this work on GitHub Pages.

### CODE REVIEW

File: packages/web/src/public/icons/icon.svg
Comment (andrewbranch) [inline packages/web/src/public/icons/icon.svg:1]: What’s the new icon for?

**Comment by andrewbranch** [comment]
Let’s try it!

File: packages/web/src/public/icons/icon.svg
Comment (johnnyreilly) [reply packages/web/src/public/icons/icon.svg:1]: I took the svg icon which is black and white and applied the TS colour to it. So there's now an SVG icon which could be used without needing to apply a colour in the HTML. It's not used as yet as I was unsure if you deliberately wanted a black and white icon?

**Comment by johnnyreilly** [comment]
Sad news

```json
headers {
  "accept": "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8",
  "connection": "keep-alive",
  "host": "48f7b04d-5fe4-4823-b66a-5176efe29bb7.azurewebsites.net",
  "user-agent": "Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36",
  "accept-language": "en-GB,en;q=0.9,de-DE;q=0.8,de;q=0.7,en-US;q=0.6",
  "authorization": "Bearer REDACTED",
  "max-forwards": "8",
  "referer": "https://www.opengraph.xyz/",
  "traceparent": "00-372fd5162dbaf01da7705b95ae3183d8-8e2eb559c996a914-00",
  "cf-connecting-ip": "92.238.72.13",
  "cf-ipcountry": "GB",
  "x-forwarded-for": "92.238.72.13, 172.71.178.16:51086, [fd00::45e1:7212:100:c0a8:fafc]:49432",
  "cf-ray": "7fda092ccefedd72-LHR",
  "x-forwarded-proto": "https",
  "cf-visitor": "{\"scheme\":\"https\"}",
  "sec-ch-ua": "\"Chromium\";v=\"116\", \"Not)A;Brand\";v=\"24\", \"Google Chrome\";v=\"116\"",
  "dnt": "1",
  "sec-ch-ua-mobile": "?1",
  "sec-ch-ua-platform": "\"Android\"",
  "sec-fetch-site": "cross-site",
  "sec-fetch-mode": "no-cors",
  "sec-fetch-dest": "image",
  "cdn-loop": "cloudflare",
  "x-arr-log-id": "78ec7568-9ae6-4d9a-932f-5d2bdc94d32f",
  "client-ip": "10.0.32.13:36152",
  "x-site-deployment-id": "48f7b04d-5fe4-4823-b66a-5176efe29bb7",
  "was-default-hostname": "48f7b04d-5fe4-4823-b66a-5176efe29bb7.azurewebsites.net",
  "x-appservice-proto": "https",
  "x-arr-ssl": "2048|256|CN=Microsoft Azure TLS Issuing CA 01, O=Microsoft Corporation, C=US|CN=*.azurewebsites.net, O=Microsoft Corporation, L=Redmond, S=WA, C=US",
  "x-forwarded-tlsversion": "1.2",
  "x-original-url": "/api/image-with-text",
  "x-waws-unencoded-url": "/api/image-with-text",
  "x-ms-original-url": "https://johnnyreilly.com/api/image-with-text",
  "x-ms-request-id": "78ec7568-9ae6-4d9a-932f-5d2bdc94d32f",
  "x-ms-auth-token": "Bearer REDACTED",
  "disguised-host": "48f7b04d-5fe4-4823-b66a-5176efe29bb7.azurewebsites.net",
  "x-ms-client-principal-id": "anonymous",
  "x-ms-client-principal-idp": "azureStaticWebApps"
}
```

I got the above from using https://arethetypeswrong.github.io/test.html?p=jQuery%401.7.4 on https://www.opengraph.xyz/url/https%3A%2F%2Farethetypeswrong.github.io%2Ftest.html%3Fp%3DjQuery%25401.7.4

No information about the https://arethetypeswrong.github.io/test.html?p=jQuery%401.7.4 surfaces. I think we can revert this PR - it's a fail I'm afraid

**Comment by andrewbranch** [comment]
Ah well, thank you for doing this experiment! I’ve been wondering about it for a long time and I figured it was unlikely to work, and now my curiosity has been satisfied.

**Comment by johnnyreilly** [comment]
No worries - I had fun doing it!

**Pull request merged**